pkg/ccl/sqlproxyccl/backend_dialer_test.go

```diff
@@ -36,12 +38,37 @@ func TestBackendDialTLSInsecure(t *testing.T) {
 	sql, _, _ := serverutils.StartServer(t, base.TestServerArgs{Insecure: true})
 	defer sql.Stopper().Stop(ctx)
 
-	conn, err := BackendDial(startupMsg, sql.ServingSQLAddr(), &tls.Config{})
+	conn, err := BackendDial(context.Background(), startupMsg, sql.ServingSQLAddr(), &tls.Config{})
 	require.Error(t, err)
 	require.Regexp(t, "target server refused TLS connection", err)
 	require.Nil(t, conn)
 }
 
+func TestBackendDialBlackhole(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	listener, err := net.Listen("tcp", "127.0.0.1:")
```

Comment:
what's the trailing `:` here? did you mean `:0` to auto-allocate a port?

Reply:
Yep! The intent was to trigger the port auto allocation behavior. I agree 0 is more obvious, so I switched it.